_posts/2019-01-23-multifile-python-functions.md

```diff
@@ -0,0 +1,213 @@
+---
+title: Multifile python functions
+description: "Tidy python in an OpenFaaS world; or, what happens when you want to split your Python function into multiple modules?"
+date: 2019-01-23
+image: /images/multifile-python-function/pexels-book-stack-books-bookshop-264635.jpg
+categories:
+  - python
+author_staff_member: lucas
+dark_background: true
+---
+
+As a core contributor to OpenFaaS, I hangout in the OpenFaaS Slack a lot, you can [join us here][openfaas-slack-singnup]. Most recently that meant trying to answer this question
+
+> [How can I] write a function in python that uses multiple source files so that the main `handler.py` is able to import helper functions from separate source files.
+
+Today, I will walk though creating exactly this kind of function by creating a word count function.
+
+**TL;DR**: Use explicit relative imports to split your python code into several files. If you want to quickly jump to the finished code, I have create a [public repo here][project-repo].
+
+I assume some familiarity with OpenFaaS and that you may have even started/completed the [OpenFaaS Workshop][workshop-repo], so I won't go into a lot of detail about the specific OpenFaaS setup instructions below. Instead, I want to focus on how to comfortably develop a Python function as it goes from a small "one-liner" function to a larger multi-module project.
+
+## Set up a python environment
+
+This project uses Python 3 and I like to use Conda for my development environments:
```

Comment:
I'm confused why we need Conda on the host and these other deps?

Reply:
Please can we remove it from the blog post - `conda` should not a requirement for working with OpenFaaS and Python. I know this is something you like in your own environment, but in my opinion it will be confusing for new people.

Reply:
I removed this